source/common/config/tls_context_json.cc

```diff
@@ -34,22 +40,27 @@ void TlsContextJson::translateCommonTlsContext(
     common_tls_context.add_alpn_protocols(alpn_protocol);
   }
 
-  common_tls_context.mutable_deprecated_v1()->set_alt_alpn_protocols(
-      json_tls_context.getString("alt_alpn_protocols", ""));
+  if (!json_tls_context.getString("alt_alpn_protocols", "").empty()) {
+    common_tls_context.mutable_deprecated_v1()->set_alt_alpn_protocols(
+        json_tls_context.getString("alt_alpn_protocols", ""));
+  }
 
-  translateTlsCertificate(json_tls_context, *common_tls_context.mutable_tls_certificates()->Add());
+  if (!json_tls_context.getString("cert_chain_file", "").empty() ||
+      !json_tls_context.getString("private_key_file", "").empty()) {
+    translateTlsCertificate(json_tls_context,
+                            *common_tls_context.mutable_tls_certificates()->Add());
+  }
```

Comment:
This one isn't necessary, since those values are checked in `translateTlsCertificate()` anyway.

Reply:
The check here is to avoid create empty TlsCertificate being passed to `translateTlsCertificate()` if neither of them exists.